### ISSUE TITLE
wide and deep code written fully in tensorflow 2.1 throws tensorflow 1 and 2 incompatability error

### ISSUE THREAD

**Issue by shnamin** [tensorflow/tensorflow#36524]
Hi, 
I am writing a code in tensorflow 2.1 with feature_columns but it is giving me errors:

This is the code snippet:
I am keeping the data in pandas dataframe: df
I am defining the feature columns similar to the code snippet below, containing crossed features, categorical features, numerical features:

cat_f1=tf.feature_column.categorical_column_with_vocabulary_list('cat_col_name_1' , df['cat_col_name_1'].unique())
...
num_fn  = tf.feature_column.numeric_column("num_col_name_n") 

wide_columns = {cat_f1.key: cat_f1,
...
}
deep_columns = {
...
num_fn.key: num_fn,
}
output_feature = ['label']

And this is how I am defining my input_fn:

def input_fn(df,inner_batch_size=-1,shuffle=False):
    input_features=[]
    for key ,_ in wide_columns.items():
        input_features.append(key)
    for key, _ in deep_columns.items():
        if '-crossed' in key:
            k = key.replace('-crossed','')
            keys=[]
            keys = k.split('-')
            for k in keys: 
                input_features.append(k)
        else: 
            input_features.append(key)
    dataset=tf.data.Dataset.from_tensor_slices((dict(df[input_features]) , df[output_feature].values))
    if shuffle ==True: 
        dataset = dataset.shuffle(100).repeat()
    if inner_batch_size>0:
        dataset = dataset.batch(inner_batch_size)    
    return dataset

And the definition of estimator: 

model_dir = tempfile.mkdtemp()
optimizer = tf.keras.optimizers.RMSprop(learning_rate=0.001)
estimator = tf.estimator.DNNLinearCombinedClassifier(model_dir = model_dir,
                                                linear_feature_columns = wide_columns , 
                                                dnn_feature_columns=deep_columns,
                                                dnn_hidden_units = [100,50],
                                                dnn_optimizer = optimizer)
 

and after data preparation, calling the train:
estimator.train(input_fn = lambda: input_fn(train_batch_data,inner_batch_size,shuffle=True) ,steps=10)
metrics = estimator.evaluate(input_fn = lambda: input_fn(val_batch_data,shuffle=False)  , steps = 5)
for key in sorted(metrics):
    print ("%s: %s" % (key, results[key]))

This throws the error:

Received a feature column from TensorFlow v1, but this is a TensorFlow v2 Estimator. Please either use v2 feature columns (accessible via tf.feature_column.* in TF 2.x) with this Estimator, or switch to a v1 Estimator for use with v1 feature columns (accessible via tf.compat.v1.estimator.* and tf.compat.v1.feature_column.*, respectively.

What is the problem?

**Comment by ravikyram** [comment]
@shnamin 

Can you please share the simple standalone code with proper indentation and supporting files or colab link to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by shnamin** [comment]
I figured out what the problem was. TF2 requires deep and wide features to be passed as lists, which is a difference between TF2.1 and TF1.X. By passing the list, the error disappears. 
There are no examples in TF2.X for wide and deep, and the examples that I found online are for TF1.X and refer to broken links. This causes confusion, I suggest writing a notebook/example for wide and deep in TF2.1 as it is still a popular approach. 

Thanks @ravikyram for the great support anyway.

**Comment by shnamin** [comment]
I was passing a dict to them, as is shown in the example.

**Comment by ravikyram** [comment]
@shnamin 

Glad to know you found the solution. Please, see the [link](https://medium.com/ml-book/demonstration-of-tensorflow-feature-columns-tf-feature-column-3bfcca4ca5c4) if it helps you. Please,let me know if we can close this issue since it looks to be fixed. Thanks!
